src/System.Private.ServiceModel/tools/SelfHostedWcfService/Program.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System;
+using System.ServiceModel;
+using System.ServiceModel.Description;
+using System.ServiceModel.Web;
+using WcfService;
+
+namespace SelfHostedWCFService
+{
+    public class SelfHostedWCFService
+    {
+        static int httpPort = 8081;
+        static int httpsPort = 44285;
+        static int tcpPort = 809;
+        static int websocketPort = 8083;
+        static int websocketsPort = 8084;
+
+        static private void GetPorts()
+        {
+            httpPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("httpPort")) ? httpPort : int.Parse(Environment.GetEnvironmentVariable("httpPort"));
+            httpsPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("httpsPort")) ? httpsPort : int.Parse(Environment.GetEnvironmentVariable("httpsPort"));
+            tcpPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("tcpPort")) ? tcpPort : int.Parse(Environment.GetEnvironmentVariable("tcpPort"));
+            websocketPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("websocketPort")) ? websocketPort : int.Parse(Environment.GetEnvironmentVariable("websocketPort"));
+            websocketsPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("websocketsPort")) ? websocketsPort : int.Parse(Environment.GetEnvironmentVariable("websocketsPort"));
+        }
+
+        static void Main()
+        {
+            GetPorts();
+
+            string httpBaseAddress = string.Format(@"http://localhost:{0}", httpPort);
+            string httpsBaseAddress = string.Format(@"https://localhost:{0}", httpsPort);
+            string tcpBaseAddress = string.Format(@"net.tcp://localhost:{0}", tcpPort);
+            string websocketBaseAddress = string.Format(@"http://localhost:{0}", websocketPort);
+            string websocketsBaseAddress = string.Format(@"https://localhost:{0}", websocketsPort);
+
+            //Do not need to catch exceptions and dispose service hosts as the process will terminate
+            Uri crlUrl = new Uri(string.Format("http://localhost/CrlService.svc", httpPort));
+            WebServiceHost host = new WebServiceHost(typeof(CrlService), crlUrl);
+            WebHttpBinding binding = new WebHttpBinding();
+            host.AddServiceEndpoint(typeof(ICrlService), binding, "");
+            ServiceDebugBehavior stp = host.Description.Behaviors.Find<ServiceDebugBehavior>();
+            stp.HttpHelpPageEnabled = false;
+            host.Open();
+
+            Uri[] utilTestServiceHostbaseAddress = new Uri[]  { new Uri(string.Format("{0}/Util.svc", httpBaseAddress))};
```

Comment:
For later... I'd like to find a way to make it easier to add new services and hosts that both self-host and IIS-host use. I know the activation models are different, but today I need to touch multiple files to add a service and host, and it's not obvious if I forgot to add an entry to the right file.  Also, I think we want to make the hosts take generic parameters, not typeof.  Something to think about after this is all checked in.